pulp_deb/app/models/content.py

```diff
@@ -327,3 +357,211 @@ class PackageReleaseComponent(Content):
+            if content_artifact:
+                if content_artifact.artifact:
+                    sha256 = content_artifact.artifact.sha256
+                    size = content_artifact.artifact.size
+                else:
+                    remote_artifact = content_artifact.remoteartifact_set.first()
+                    sha256 = remote_artifact.sha256
+                    size = remote_artifact.size
+                # Sha256 is required so better to not filter out incomplete data
+                contents.append(
+                    {
+                        "name": os.path.basename(content_artifact.relative_path),
+                        "sha256": sha256,
+                        "size": size,
+                    }
+                )
+        return contents
+
+    @property
+    def checksums_sha512(self):
+        """Generate 'Checksums-Sha512' list from content artifacts."""
+        contents = []
+        for content_artifact in self.contentartifact_set.all():
+            if content_artifact:
+                if content_artifact.artifact:
+                    sha512 = content_artifact.artifact.sha512
+                    size = content_artifact.artifact.size
+                else:
+                    remote_artifact = content_artifact.remoteartifact_set.first()
+                    sha512 = remote_artifact.sha512
+                    size = remote_artifact.size
+                # Sha512 is optional so filter out incomplete data
+                if sha512 is not None:
+                    contents.append(
+                        {
+                            "name": os.path.basename(content_artifact.relative_path),
+                            "sha512": sha512,
+                            "size": size,
+                        }
+                    )
+        return contents
+
+    @property
+    def files(self):
+        """Generate 'Files' list from content artifacts."""
+        contents = []
+        for content_artifact in self.contentartifact_set.all():
+            if content_artifact:
+                if content_artifact.artifact:
+                    md5 = content_artifact.artifact.md5
+                    size = content_artifact.artifact.size
+                else:
+                    remote_artifact = content_artifact.remoteartifact_set.first()
+                    md5 = remote_artifact.md5
+                    size = remote_artifact.size
+                # md5 is required so better to not filter out incomplete data
+                contents.append(
+                    {
+                        "name": os.path.basename(content_artifact.relative_path),
+                        "md5sum": md5,
+                        "size": size,
+                    }
+                )
+        return contents
+
+    repo_key_fields = ("source", "version")
+
+    class Meta:
+        default_related_name = "%(app_label)s_%(model_name)s"
+
+
+class DscFileReleaseComponent(Content):
```

Comment:
```suggestion
class SourcePackageReleaseComponent(Content):
```
I would also rename this here.

Reply:
Same here, I will look at renaming it.

Reply:
Rename has been done.